### PR TITLE
Update README URLs based on HTTP redirects

### PR DESCRIPTION
**Pull request by ReadmeCritic**
Created with https://github.com/dkhamsing/frankenstein
### HTTPS Corrected URLs

| Was | Now |
| --- | --- |
| http://godoc.org/github.com/cockroachdb/cockroach/kv | https://godoc.org/github.com/cockroachdb/cockroach/kv |
| http://godoc.org/github.com/cockroachdb/cockroach/server | https://godoc.org/github.com/cockroachdb/cockroach/server |
| http://godoc.org/github.com/cockroachdb/cockroach/storage | https://godoc.org/github.com/cockroachdb/cockroach/storage |
| http://img.shields.io/badge/status-alpha-red.svg | https://img.shields.io/badge/status-alpha-red.svg |
### Other Corrected URLs

| Was | Now |
| --- | --- |
| http://www.youtube.com/watch?v=MEAuFgsmND0&feature=youtu.be | https://www.youtube.com/watch?feature=youtu.be&v=MEAuFgsmND0 |
| https://code.google.com/p/leveldb/ | https://github.com/google/leveldb |
| https://docs.docker.com/installation/mac/#from-your-shell | http://docs.docker.com/engine/installation/mac/#from-your-shell |

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/3406)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,7 +37,7 @@ CockroachDB is currently in alpha. See our
 
 #### Using Docker
 
-Install Docker! On OSX ([official docs](https://docs.docker.com/installation/mac/#from-your-shell)):
+Install Docker! On OSX ([official docs](http://docs.docker.com/engine/installation/mac/#from-your-shell)):
```
Comment (tamird) [inline README.md]: This should stay https; they have a broken redirect

Comment (ReadmeCritic) [inline README.md]: Ok

**Comment by ReadmeCritic** [comment]
@tamird updated

**Comment by tamird** [comment]
Thanks!

**Comment by ReadmeCritic** [comment]
:smile:

**Pull request merged**